contrib/internal/httptrace/httptrace.go

```diff
@@ -0,0 +1,62 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+// Package httptrace provides functionalities to trace HTTP requests that are commonly required and used across
+// contrib/** integrations.
+package httptrace
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+// StartRequestSpan starts an HTTP request span with the standard list of HTTP request span tags. URL query parameters
+// are added to the URL tag when queryParams is true. Any further span start option can be added with opts.
+func StartRequestSpan(r *http.Request, service, resource string, queryParams bool, opts ...ddtrace.StartSpanOption) (tracer.Span, context.Context) {
```

Comment:
I have some ideas on simplifying this function signature and as a result the API.

### service & resource

While these two are mandatory, there is nothing in the function stopping us from giving an empty string. For that reason, I think it'd be much nicer if we removed both service & resource and instead just use `opts` because they already allow setting both of these. Additionally, perhaps it makes sense to log a message (at the "warn" or "debug" level) if service and/or resource are empty.

### queryParams

I would like to propose removing `queryParams` from here for multiple reasons:
* I think that bool args make for an ugly API even if that API is internal.
* There are way too many args on this function, contributing to the ugliness of its API (not your fault, it came like this, but let's improve it!)
* It is `false` 99% of the time, except for `net/http`

I'd propose to just remove it, and inside the `net/http` package, when the query params are needed, simply use `tracer.Tag(ext.HTTPURL, r.URL.Path + "?" + r.URL.RawQuery)`.

WDYT?

Reply:
- service & resource: I don't see any tracer option to set the resource name. As for the service name, there's a side effect in the option setting the global service name.  Do you still think I should use it? What about the resource name?

- Query params: we can especially remove it now that we know we can have them by default.

Reply:
Oh, they are there: https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/option.go#L746-L755. They are actually used in this very function 🙂

Reply:
Done in https://github.com/DataDog/dd-trace-go/pull/1286/commits/95a7b98715210af126f661f33e3d65ca25a3e66e

Note that we could go further by reworking the signature of `TraceAndServe` but the change got too big for the original purpose of this pull request which is about adding the new user agent tag to every HTTP integration.